### PR TITLE
Added documentation for REDIS optimistic lock system.

### PR DESCRIPTION
**Pull request by Fl4m3Ph03n1x**
### Description of change

Documentation update for optimistic locks using WATCH.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -704,6 +704,99 @@ significantly compared to firing the same commands in a loop without waiting for
...
+Using `multi` you can make sure your modifications run as a transaction, but you 
+can't be sure you got there first. What if another client modified a key while 
+you were working with it's data?
+
+To solve this, Redis supports the [WATCH](https://redis.io/topics/transactions) 
+command, which is meant to be used with MULTI:
+
+```js
+var redis  = require("redis"),
+    client = redis.createClient();
```
Comment (stockholmux) [inline README.md]: Please add something to the arguments of `redis.createClient`, even if it's just an empty object indicating that something _should_ be there. No arguments means that you have no password on your Redis instance and it's a huge security issue, so we always try to illustrate best practices.

File: README.md [hunk 2]
```diff
@@ -704,6 +704,99 @@ significantly compared to firing the same commands in a loop without waiting for
...
+        clients.alterer.quit();
+      });
+  },1000);
+});
+```
+
+**NOTE**: Redis WATCH does not work on fields of hashes and other objects. You can watch a hash
+to see if anything inside it was modified, but you cannot watch a specific hash field for a 
+modification.
+
```
Comment (stockholmux) [inline README.md]: A better way to put this is that Redis WATCH works on whole _keys_ - you can also mention the hash fields as it's a common misconception by beginners.

Maybe also mention that you can't watch individual set members, that way it is a bit more general and drives home the point that it's the whole _key's value_ being watched, no matter the type of the key.

**Comment by stockholmux** [comment]
A few small nits but otherwise awesome. Make the changes and I'll merge. Thanks for your contribution!

**Comment by Fl4m3Ph03n1x** [comment]
Alright, did the requested improvements. Let me know if everything is to your liking!

File: README.md (hunk 2 above)
Comment (Fl4m3Ph03n1x) [reply README.md]: Done :D

**Comment by stockholmux** [comment]
Thanks for your contribution!

**Comment by Fl4m3Ph03n1x** [comment]
My pleasure, 
Only wish I could contribute more with more examples and snippets. 

All my projects have a project page and I think it would benefit this one also a lot, but I also understand your desire to keep documentation as minimal and necessary as possible to not replicate the already good documentation Redis has. 

Oh well, time to contribute to MDN and update my SO answer :P

**Comment by funny-falcon-at-joomcode** [comment]
https://redis.io/topics/transactions
> Redis scripting and transactions
> A Redis script is transactional by definition, so everything you can do with a Redis transaction, you can also do with a script, and usually the script will be both simpler and faster.

**Pull request merged**